django/db/models/fields/tuple_lookups.py

```diff
@@ -301,7 +307,7 @@ def check_rhs_elements_length_equals_lhs_length(self):
             )
 
     def check_rhs_is_query(self):
```

Comment:
Do you think the function name and error message should be updated here?

Reply:
I think _query_ is good enough as whether `Query` or `Subquery` will be received is an implementtion detail.

[As discussed in the other PR](https://github.com/django/django/pull/19108#discussion_r1932339414) I think there is room for cleaning this whole `QuerySet`, `Query`, and  `Subquery` situation eventually so I think that keeping the generic _query_ term for now makes sense.